### ISSUE TITLE
Notice: unserialize(): Error at offset 0 of 5264 bytes

### ISSUE THREAD

**Issue by mindfriction** [Nosto/nosto-magento2#655]
**Magento 2 version(s) used**: 2.3.3
**Extension version(s) affected**: 4.0.4

**Description**  
Updated to the latest version 4.0.4 and I'm now seeing many of the following errors in Magento system.log:
[2020-02-25 03:57:15] nosto.CRITICAL: Notice: unserialize(): Error at offset 0 of 5280 bytes in /chroot/home/xxx/xxxx.com/html/vendor/nosto/module-nostotagging/Model/Service/Product/DefaultProductSerializer.php on line 52 [] []
[2020-02-25 03:57:15] nosto.CRITICAL: Notice: unserialize(): Error at offset 0 of 5264 bytes in /chroot/home/xxx/xxxx.com/html/vendor/nosto/module-nostotagging/Model/Service/Product/DefaultProductSerializer.php on line 52 [] []

**How to reproduce**  
composer update to nosto/module-nostotagging version 4.0.4 from 4.0.3

**Magento 2 mode**
 Production

**Full page cache**
Enabled 

**Nosto indexer mode**
 Update By Schedule

**Comment by hannunosto** [comment]
Hi @mindfriction, did you run `bin/magento setup:upgrade` command after composer update? That will convert the serialized data to base 64 encoded string. Can you also check if the data in `nosto_tagging_product_cache` table is base 64 encoded, empty or still in serialized format and let us know.

**Comment by mindfriction** [comment]
Hi @hannunosto - thank you for responding so quickly!

Yes, I did run bin/magento setup:upgrade after the composer update, but I continued to get these errors from the time at which bin/magento setup:upgrade completed at 2020-02-25 03:05:15 until 2020-02-25 08:32:16 which was the last "Notice: unserialize()" in the logs. I have confirmed that all of the data in nosto_tagging_product_cache table is base 64 encoded now. 

However the updated_at timestamps for around half of the products in this table are 2020-02-25 05:00:42 (12:00am EST) then for the other half of products all have updated_at timestamps 2020-02-25 09:00:12 (4:00am EST).

So this leads me to believe that the serialized data wasn't actually converted at the time of the setup:upgrade, but either by a cron that kicked off at 12:00am EST and then again at 4:00am EST instead. I'm just not certain if this was a Nosto cron, Magento Indexer cron or a scheduled sync/update from Nosto?

I should mention that in my Nosto/Magento settings I have:
"Build product data cache in cron" set to NO
(Set this yes if you want the cached product data to be built in a cron job instead of in Magento indexer)

I have a feeling that this setting is the why the nosto_tagging_product_cache table didn't update until 12:00am and 4:00am. Either way, the issue is resolved but shouldn't the data have converted at the time of setup:upgrade, regardless? Or no...?

Thanks!
